addons/point_of_sale/models/pos_payment_method.py

```diff
@@ -64,13 +60,24 @@ def _compute_open_session_ids(self):
         for payment_method in self:
             payment_method.open_session_ids = self.env['pos.session'].search([('config_id', 'in', payment_method.config_ids.ids), ('state', '!=', 'closed')])
 
-    @api.onchange('is_cash_count')
-    def _onchange_is_cash_count(self):
-        if not self.is_cash_count:
-            self.cash_journal_id = False
-        else:
+    @api.depends('journal_id', 'split_transactions')
+    def _compute_type(self):
+        for pm in self:
+            if pm.journal_id.type in {'cash', 'bank'}:
+                pm.type = pm.journal_id.type
+            else:
+                pm.type = 'pay_later'
+
+    @api.onchange('journal_id')
+    def _onchange_journal_id(self):
+        if self.is_cash_count:
             self.use_payment_terminal = False
 
+    @api.depends('type')
+    def _compute_is_cash_count(self):
```

Comment:
Ok, I've just seen it, did you keep it to avoid to much modifications? or some other reason. I am ok to merge with it if we remove it later

Reply:
I kept 'is_cash_count' because it has more references than I expected. I guess we can clean it up later if you agree.